src/libecalc/common/utils/rates.py

```diff
@@ -624,6 +624,56 @@ def resample(self, freq: Frequency) -> TimeSeriesIntensity:
         )
 
 
+class TimeSeriesStreamDayRate(TimeSeriesFloat):
+    """
+    Domain/core? layer only
+
+    Explicit class for only internal core usage. Makes it easy to catch that the
+    type of data is rate, and has been converted to Stream Day for internal usage.
+
+    When used internally, rate is handled as a "point in time float". It is only needed to
+    be handled specifically when reporting, e.g. converting to calendar day rate, if needed.
+
+    Will need to make a decision whether it makes sense to also represent and convert to calendar day rate.
+    TODO: Remove timeseries internally...use lists directly or some other timestep agnostic thing?
+    models have no knowledge of timesteps, but components must have since they merge results...so might be
+    a thing in the "component layer"...which is more or less network layer?
+    """
+
+    ...
+
+    def to_timeseries_rate(self, regularity: List[float]) -> TimeSeriesRate:
+        # TODO: Verify that length of regularity is the same, or 1
+        return TimeSeriesRate(
+            timesteps=self.timesteps,
+            values=self.values,
+            unit=self.unit,
+            rate_type=RateType.STREAM_DAY,
+            regularity=regularity,
+        )
+
+
+class TimeSeriesCalendarDayRate(TimeSeriesFloat):
```

Comment:
Probably not needed, was intended for HCExport etc that only resides in application layer, but then we can use the old TimeSeriesRate class (under a new name?)